### DISCUSSION TITLE
Stream-related internal events

### DISCUSSION THREAD

**Discussion by ccarnero** [rabbitmq/rabbitmq-server#10946] [Questions] [answered]
does _rabbitmq_event_exchange_ plugin support **streams** event create/delete notifications?

Thanks, Christian

**Comment by cihatsolak** [comment]
The rabbitmq_event_exchange plugin in RabbitMQ doesn't inherently provide stream event notifications such as creation or deletion. Its main purpose is to route internal RabbitMQ events to designated exchanges for client consumption. For stream-related notifications, you can utilize RabbitMQ's Management Plugin through its HTTP API or CLI commands. Custom solutions or integration with external monitoring tools may be needed for more specific event notifications within RabbitMQ.

**Comment by michaelklishin** [comment]
There is no reason why [internal events](https://www.rabbitmq.com/docs/logging#internal-events) cannot include certain events for streams, e.g. when they are created and deleted. Not that highly dynamic streams is a workflow streams were designed for but as far as auditing goes, it would be a great addition.

You are welcome to contribute a PR that would emit relevant events when a stream is created or deleted. Using the `queue.declared` (`queue_declared` internally) and `queue.deleted` (`queue_deleted`) events as examples would help. It should be a fairly trivial addition with only one important caveat: we should emit this event once on the leader replica, not on every replica.

**Reply by ccarnero** [reply]
I really appreciate your confidence. I have quite a bit of programming experience, but Erlang is new to me. I started by looking at the file rabbit_exchange_type_event.erl to get an idea, but honestly, there might be someone more suited for this specific challenge. Either way, I appreciate the time you've taken to respond! thanks!

**Comment by dcorbacho** [comment, marked as answer]
Stream creation and deletion events are already emitted, they are tagged as any other queue type: `queue.created` and `queue.deleted`
